datastore/save.go

```diff
@@ -53,6 +54,32 @@ func reflectFieldSave(props *[]Property, p Property, name string, opts saveOpts,
 	switch x := v.Interface().(type) {
 	case *Key, time.Time, GeoPoint:
 		p.Value = x
+	case civil.Date, civil.DateTime, civil.Time:
+		if dateVal, ok := v.Interface().(civil.Date); ok {
+			p.Value = dateVal.In(time.Now().Location())
+			*props = append(*props, p)
+			return nil
+		}
+		if timeVal, ok := v.Interface().(civil.Time); ok {
+			var format string
+			if timeVal.Nanosecond == 0 {
+				format = "15:04:05"
+			} else {
+				format = "15:04:05.000000000"
+			}
+			val, err := time.Parse(format, timeVal.String())
+			if err != nil {
+				return fmt.Errorf("datastore: error while parsing time: %v", err)
+			}
+			p.Value = val
+			*props = append(*props, p)
+			return nil
+		}
+		if dateTimeVal, ok := v.Interface().(civil.DateTime); ok {
```

Comment:
I don't see a test for this case, could you add one?

Reply:
+1.

Reply:
There is a case: https://github.com/googleapis/google-cloud-go/pull/3202/files#r522815770
Still, there was a copy-past field names error. I've fixed it.